pkg/registry/core/shoot/strategy.go

```diff
@@ -65,6 +66,9 @@ func (shootStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 
 	shoot.Generation = 1
 	shoot.Status = core.ShootStatus{}
+
+	// Defaults spec.kubernetes.allowPrivilegedContainers=false for k8s version >= 1.25 and true for lower versions
+	defaultAllowPrivilegedContainers(shoot)
```

Comment:
Why are we defaulting here in the `registry` package and not in the `defaults.go`?

Reply:
Oops. Somewhere in my mind , It was there that we had to do this only for update calls, so chose not to do it in `defaults.go`. But turns out that's not the case.